core/src/main/scala/org/apache/spark/network/nio/NioBlockTransferService.scala

```diff
@@ -143,7 +143,7 @@ final class NioBlockTransferService(conf: SparkConf, securityManager: SecurityMa
       level: StorageLevel)
     : Future[Unit] = {
     checkInit()
-    val msg = PutBlock(blockId, blockData.nioByteBuffer(), level)
+    val msg = PutBlock(blockId, blockData.nioByteBuffer().firstByteBuffer(), level)
```

Comment:
`NioBlockTransferService` is totally broken with these changes, this is just to make it compile.